ra_aid/tools/expert.py

```
@@ -142,61 +266,67 @@ def ask_expert(question: str) -> str:
-    console.print(
-        Panel(Markdown(response.content), title="Expert Response", border_style="blue")
-    )
+        # Format and display response
+        console.print(
+            Panel(Markdown(response.content), title="Expert Response", border_style="blue")
+        )
 
-    return response.content
+        return response.content
+    except Exception as e:
+        raise ToolExecutionError(f"Expert consultation failed: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)